File: backend/app/schemas/workflow.py
from pydantic import BaseModel, Field


class WorkflowCreateRequest(BaseModel):
    prompt: str = Field(min_length=10, max_length=1000)
    output_format: str = Field(default="xsoar", pattern="^(xsoar|splunk|both)$")


class WorkflowResponse(BaseModel):
    name: str
    structured_logic: dict
    playbook_steps: list[dict]
    branches: list[dict]
    integrations: list[str]
    api_calls: list[dict]
    error_handling: list[dict]
    approvals: list[dict]
    escalation: dict
    xsoar_json: dict
    splunk_json: dict | None
    visual_flow: dict


class WorkflowExportResponse(BaseModel):
    workflow_id: int
    format: str
    content: str

File: frontend/src/types/workflow.ts
export interface WorkflowResponse {
  name: string
  structured_logic: Record<string, unknown>
  playbook_steps: { title: string; action: string }[]
  branches: { if: string; then: string; else?: string }[]
  integrations: string[]
  api_calls: { service: string; method: string; path: string }[]
  error_handling: Record<string, unknown>[]
  approvals: Record<string, unknown>[]
  escalation: Record<string, unknown>
  xsoar_json: Record<string, unknown>
  splunk_json?: Record<string, unknown>
  visual_flow: { nodes: { id: string; label: string }[]; edges: { source: string; target: string }[] }
}
